azure-pipelines.yml

```diff
@@ -39,6 +41,22 @@ steps:
     pushTag: '$(ImageVersion)'
     autoCreateRepository: true
 
+# ECRPushImage does not support specifying multiple tags at push time. We therefore push a second time to
+# additionally tag the image with the 'latest' tag. Note that this does not create a second image, and
+# this push completes very quickly since it just adds an additional tag to the image.
+- task: ECRPushImage@1
```

Comment:
Maybe this is the reason we aren't using this task in our other pipelines. Do you think it would be more clear to perform the image push the same way we do in our other pipelines?

https://ni.visualstudio.com/DefaultCollection/DevCentral/_git/Skyline?path=/Web/Workspaces/Grafana/azure-pipeline.yml&version=GBmaster&line=210&lineEnd=211&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
Yeah, I think we shouldn't be different unless we have a reason to be. I don't know why this pipeline is currently using `ECRPushImage`. I didn't realize this pipeline was unique in that respect, or I would have raised an issue when the pipeline was originally created. In this case, I couldn't get the task we use in our other pipelines to work (see [this error here when I tried to use the Docker2 task](https://ni.visualstudio.com/DevCentral/_build/results?buildId=2197057&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=0febf08e-6fac-5ca3-6c4b-78e0cd0a0a95&l=44)). Perhaps that's why we're using `ECRPushImage`?

Reply:
I talked with Carson about this previously. He used this task because it seemed like a more first-class way of pushing to ECR, but acknowledged the limitation regarding multiple tags. I'm assuming the other task is failing because we need to login first. If you look at the pipeline I linked there are some steps prior to the push you probably need to pull in as well.

Reply:
Gotcha, I'll try adding the login tasks.

Reply:
Unfortunately it [still fails](https://ni.visualstudio.com/DevCentral/_build/results?buildId=2201186&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=347de3ba-75e7-5260-17f8-800303757031&l=41) even after adding the docker hub login tasks. I'm guessing it has something to do with how we're not running these tasks on the skyline build agents, but are instead using the DevCentral-Docker agents. I suspect it isn't a tricky thing to fix, but given our schedule constraints, I'd like to propose just sticking to the `ECRPushImage` task that the pipeline is already using.

I think it makes sense create some tech debt for making this pipeline look like our other pipelines and replacing the `ECRPushImage` tasks with what we do everywhere else.

Reply:
Works for me. I agree, it likely is because we're in a different agent pool. I'll go ahead and count this as resolved.